Microsoft.HealthVault.Fhir/Transformers/WeightToFhir.cs

```diff
@@ -8,19 +8,31 @@
 
 using Hl7.Fhir.Model;
 using Microsoft.HealthVault.Fhir.Constants;
+using Microsoft.HealthVault.Fhir.Vocabularies;
 using Microsoft.HealthVault.ItemTypes;
+using Microsoft.HealthVault.Thing;
 
 namespace Microsoft.HealthVault.Fhir.Transformers
 {
+    static partial class ThingBaseToFhir
+    {
+        internal static WeightToFhir s_weightToFhir = new WeightToFhir();
+
+        // Register the type on the generic ThingToFhir partial class
+        public static Observation ToFhir(this Weight weight)
+        {
+            return s_weightToFhir.ToFhirInternal(weight, ThingBaseToFhir.ToFhirInternal(weight));
+        }
+    }
+
     /// <summary>
     /// An extension class that transforms HealthVault weight data types into FHIR Observations
     /// </summary>
-    public static class WeightToFhir
+    internal class WeightToFhir
     {
-        public static Observation ToFhir(this Weight weight)
+        internal Observation ToFhirInternal(Weight weight, Observation observation)
         {
-            var observation = ThingBaseToFhir.ToFhir(weight);
-
+            observation.Category = new System.Collections.Generic.List<CodeableConcept>() { FhirCategories.VitalSigns };
```

Comment:
Yes but we will add that once we are on the server. Right now I am not adding patient information to the observations yet. 

---
In reply to: [124081208](https://github.com/Microsoft/healthvault-fhir-library/pull/3#discussion_r124081208) [](ancestors = 124081208)